src/resolver.ts

```diff
@@ -35,7 +36,8 @@ export class Resolver {
   }
 
   public resolve(source: any, opts: Types.IResolveOpts = {}): Promise<Types.IResolveResult> {
-    const runner = new ResolveRunner(source, {
+    const graph = new RefGraph<string>({ circular: true });
```

Comment:
Let's make dependency graph computation optional, or allow a custom implementation to be provided.
Studio is unlikely to need it for now, and it'll introduce an extra performance cost we cannot really afford, as we already struggle to process files with plenty of $refs.
cc @marbemac

Reply:
I think this is OK - performance impact should be negligible, and the plan as a next step is to remove the internal DepGraph instances crawlers are creating, and just use this one root DepGraph - this should help simplify many things.